component/block_cache/block_cache.go

```diff
@@ -68,36 +68,38 @@ import (
+	blockSize         uint64          // Size of each block to be cached
+	memSize           uint64          // Mem size to be used for caching at the startup
+	tmpPath           string          // Disk path where these blocks will be cached
+	diskSize          uint64          // Size of disk space allocated for the caching
+	diskTimeout       uint32          // Timeout for which disk blocks will be cached
+	workers           uint32          // Number of threads working to fetch the blocks
+	prefetch          uint32          // Number of blocks to be prefetched
+	diskPolicy        *tlru.TLRU      // Disk cache eviction policy
+	blockPool         *BlockPool      // Pool of blocks
+	threadPool        *ThreadPool     // Pool of threads
+	fileLocks         *common.LockMap // Locks for each file_blockid to avoid multiple threads to fetch same block
+	fileNodeMap       sync.Map        // Map holding files that are there in our cache
+	maxDiskUsageHit   bool            // Flag to indicate if we have hit max disk usage
+	noPrefetch        bool            // Flag to indicate if prefetch is disabled
+	prefetchOnOpen    bool            // Start prefetching on file open call instead of waiting for first read
+	enableRandomWrite bool            // Enable random write in block cache
 
 	lazyWrite    bool           // Flag to indicate if lazy write is enabled
 	fileCloseOpt sync.WaitGroup // Wait group to wait for all async close operations to complete
 }
 
 // Structure defining your config parameters
 type BlockCacheOptions struct {
-	BlockSize      float64 `config:"block-size-mb" yaml:"block-size-mb,omitempty"`
-	MemSize        uint64  `config:"mem-size-mb" yaml:"mem-size-mb,omitempty"`
-	TmpPath        string  `config:"path" yaml:"path,omitempty"`
-	DiskSize       uint64  `config:"disk-size-mb" yaml:"disk-size-mb,omitempty"`
-	DiskTimeout    uint32  `config:"disk-timeout-sec" yaml:"timeout-sec,omitempty"`
-	PrefetchCount  uint32  `config:"prefetch" yaml:"prefetch,omitempty"`
-	Workers        uint32  `config:"parallelism" yaml:"parallelism,omitempty"`
-	PrefetchOnOpen bool    `config:"prefetch-on-open" yaml:"prefetch-on-open,omitempty"`
+	BlockSize         float64 `config:"block-size-mb" yaml:"block-size-mb,omitempty"`
+	MemSize           uint64  `config:"mem-size-mb" yaml:"mem-size-mb,omitempty"`
+	TmpPath           string  `config:"path" yaml:"path,omitempty"`
+	DiskSize          uint64  `config:"disk-size-mb" yaml:"disk-size-mb,omitempty"`
+	DiskTimeout       uint32  `config:"disk-timeout-sec" yaml:"timeout-sec,omitempty"`
+	PrefetchCount     uint32  `config:"prefetch" yaml:"prefetch,omitempty"`
+	Workers           uint32  `config:"parallelism" yaml:"parallelism,omitempty"`
+	PrefetchOnOpen    bool    `config:"prefetch-on-open" yaml:"prefetch-on-open,omitempty"`
+	EnableRandomWrite bool    `config:"enable-random-write" yaml:"enable-random-write,omitempty"`
```

Comment:
What does actually --block-cache-enable-random-write mean for user?

if I make the above flag as false, what happens to the following conditions allowed/not allowed?
1. user can write sequentially from 0th offset to the end of the file, without allowing seeking in the file.
2. user can start seek to some offset and start writing sequentially from there.

Reply:
If the flag is false, the random write will be blocked in the following scenarios
1. The block where the write call comes is not in the buffer list (both cooking and cooked)
2. Write call comes for the block which is not the last block
3. The block is present in the blockIdList map indicating it was staged earlier and deleted from the buffer list

For your case,
1. User can seek back if the block has not been deleted from the buffer list
2. This is allowed as per the above conditions